apps/verticalhome/js/pin_apps.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+(function (exports) {
+
+  function PinAppItem(index) {
+    if (index === null || index === undefined){
+      return false;
+    }
+    this.index = index;
+    //creating pin app element
+    var pinElem = document.createElement('div');
+    pinElem.className = 'pin-app-item';
+    pinElem.setAttribute('data-index', this.index);
+    pinElem.innerHTML = "<img class='pin-app-icon'><br></img>" +
+            "<span class='title'></span>";
+    var pinList = document.getElementById('pin-apps-list');
+    var moreAppsLi = document.getElementById('moreApps');
+    pinList.insertBefore(pinElem, moreAppsLi);
```

Comment:
Is data-index meant to match the index of the element in the pinList.childElements[] array? If so, you probably don't need to pass the index and can calculate it....